### PR TITLE
Fix group_id handling for all access control resources

### LINKED ISSUE #950: wrong vcd_org_vdc_access_control group_id
After re-applying a `vcd_org_vdc_access_control` configuration, the resource is always updated in-place. This happens because the group_id is applied using the group urn e.g. `"urn:vcloud:group:6811eef7-533e-4561-a663-9f84cd0d313a"`, whereas the read only gets the ID `"6811eef7-533e-4561-a663-9f84cd0d313a"`.

### Terraform Version
Terraform v1.3.6

### Affected Resource(s)
- vcd_org_vdc_access_control

### Terraform Configuration Files

```hcl
resource "vcd_org_group" "main" {
  org           = var.org_name
  provider_type = "SAML"
  name          = vcd_org_vdc.main.name
  role          = "Defer to Identity Provider"
}

resource "vcd_org_vdc_access_control" "main" {
  org                  = var.org_name
  vdc                  = vcd_org_vdc.main.name
  shared_with_everyone = false

  shared_with {
    group_id     = vcd_org_group.main.id
    access_level = "ReadOnly"
  }
}
```

### Expected Behavior
No Update should take place

### Actual Behavior
The `vcd_org_vdc_access_control` will be updated in-place with the same group id.

```shell
  ~ resource "vcd_org_vdc_access_control" "main" {
        id                   = "urn:vcloud:vdc:f4f9c6a9-db42-46cb-a55d-45ba970ec6a3"
        # (3 unchanged attributes hidden)

      - shared_with {
          - access_level = "ReadOnly" -> null
          - group_id     = "6811eef7-533e-4561-a663-9f84cd0d313a" -> null
          - subject_name = "testvdc" -> null
        }
      + shared_with {
          + access_level = "ReadOnly"
          + group_id     = "urn:vcloud:group:6811eef7-533e-4561-a663-9f84cd0d313a"
          + subject_name = (known after apply)
        }
    }
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. `terraform plan`

### PR DESCRIPTION
**Pull request by Didainius**
Closes #950

All access control resources have a bug when using `group_id` references. They set the ID in `URN` format (`e.g. urn:vcloud:vapp:aecb5ea0-6a9a-401a-95f6-dcb402b36620`), but during read operation store it as `UUID` only (`aecb5ea0-6a9a-401a-95f6-dcb402b36620`).

This causes Terraform to report diff for each operation. 

Impacted access control resources:
* [vcd_catalog_access_control](https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/catalog_access_control)
* [vcd_org_vdc_access_control](https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/org_vdc_access_control)
* [vcd_vapp_access_control](https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/vapp_access_control)

This PR adds a new test to trigger the bug (commit 64519c0) and adds a fix for it in commit b9e611a. The fix itself is quite trivial - to add `urn:vcloud:vapp:` prefix for read operations.

### CODE REVIEW

File: .changes/v3.8.2/963-bug-fixes.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Fix a bug that caused inconsistent plan when using `group_id` in `vcd_catalog_access_control`,
+  `vcd_org_vdc_access_control`, and `vcd_vapp_access_control` resources [GH-963]
```
Comment (lvirbalas) [inline .changes/v3.8.2/963-bug-fixes.md:2]:
```suggestion
  `vcd_org_vdc_access_control` and `vcd_vapp_access_control` resources [GH-963]
```

**Review by lvirbalas — APPROVED** [review]
Just one suggestion for the changelog.

**Review by adambarreiro — APPROVED** [review]
LGTM

File: .changes/v3.8.2/963-bug-fixes.md (hunk 1 above)
Comment (Didainius) [reply .changes/v3.8.2/963-bug-fixes.md]: Fixed.

File: vcd/resource_vcd_any_access_control_test.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+// TestAccVcdAnyAccessControlGroups checks that vcd_org_vdc_access_control, vcd_vapp_access_control,
+// and vcd_catalog_access_control resources are capable of correctly handling group IDs correctly.
```
Comment (dataclouder) [inline vcd/resource_vcd_any_access_control_test.go:17]:
```suggestion
// and vcd_catalog_access_control resources are capable of handling group IDs correctly.
```
double use of "correctly"

**Review by dataclouder — APPROVED** [review]
LGTM. Just a tiny change needed

**Pull request merged**